openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java

```diff
@@ -275,19 +290,18 @@ private void fillEdgesUpward(MultiTreeSPEntry currEdge, PriorityQueue<MultiTreeS
 						_msptItem = currEdge.getItem(i);
 						_entryWeight = _msptItem.weight;
 						
-						if (_entryWeight == 0.0)
+						if (_entryWeight == Double.POSITIVE_INFINITY)
 							continue;
 
-						_msptSubItem = ee.getItem(i);
-
 						if (_msptItem.update == false) {
-		//					upwardItersSkipped++;
+							//					upwardItersSkipped++;
 							continue;
 						}
+						_msptSubItem = ee.getItem(i);
 						
 						_tmpWeight = _edgeWeight + _entryWeight;
 
-						if (_msptSubItem.weight > _tmpWeight || _msptSubItem.weight == 0.0) {
+						if (_msptSubItem.weight > _tmpWeight || _msptSubItem.weight == Double.POSITIVE_INFINITY) {
```

Comment:
I think just `_msptSubItem.weight > _tmpWeight` is necessary

Reply:
True!